src/components/flame-graph/Canvas.js

```diff
@@ -331,29 +190,24 @@ class FlameGraphCanvas extends React.PureComponent<Props> {
           thread.funcTable.name[funcIndex]
         );
 
-        const stackType = getStackType(thread, funcIndex);
+        const isSelected = selectedCallNodeIndex === callNodeIndex;
         const isHovered =
           hoveredItem &&
           depth === hoveredItem.depth &&
           i === hoveredItem.flameGraphTimingIndex;
-        let background, foreground;
-
-        if (isHovered) {
-          background = getHoverBackgroundColor(stackType);
-          foreground = '#ffffff';
-        } else {
-          background = getBackgroundColor(
-            stackType,
-            stackTiming.selfTimeRelative[i]
-          );
-          foreground = getForegroundColor(
-            stackType,
-            stackTiming.selfTimeRelative[i]
-          );
-        }
 
-        const isSelected = selectedCallNodeIndex === callNodeIndex;
-        ctx.fillStyle = isSelected ? 'Highlight' : background;
+        const categoryIndex = callNodeTable.category[callNodeIndex];
+        const category = categories[categoryIndex];
+        const colorStyles = this._mapCategoryColorNameToStyles(category.color);
+
+        const background =
+          isHovered || isSelected
+            ? colorStyles.selectedFillStyle
+            : colorStyles.unselectedFillStyle;
+        const foreground =
+          isHovered || isSelected ? colorStyles.selectedTextColor : '#000';
```

Comment:
Below, `'HighlightText'` is still used for the foreground if an item is selected, causing bad contrast on the yellow boxes:
![screenshot_2019-01-16 perf html](https://user-images.githubusercontent.com/1911402/51278922-34bfdf00-19dc-11e9-9955-9d7913e544eb.png)

Reply:
Woops, that's definitely a mistake. I meant to use the shared styling. Thanks for finding it.